modules/tensor_mechanics/src/materials/RecomputeRadialReturnIsotropicPlasticity.C

```diff
@@ -0,0 +1,166 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "RecomputeRadialReturnIsotropicPlasticity.h"
+
+template<>
+InputParameters validParams<RecomputeRadialReturnIsotropicPlasticity>()
+{
+  InputParameters params = validParams<RecomputeRadialReturnStressIncrement>();
+
+  // Linear strain hardening parameters
+  params.addParam<FunctionName>("yield_stress_function", "Yield stress as a function of temperature");
+  params.addParam<Real>("yield_stress", 0.0, "The point at which plastic strain begins accumulating");
+  params.addParam<FunctionName>("hardening_function", "True stress as a function of plastic strain");
+  params.addParam<Real>("hardening_constant", 0.0, "Hardening slope");
+  params.addCoupledVar("temp", "Coupled Temperature");
+
+  return params;
+}
+
+
+RecomputeRadialReturnIsotropicPlasticity::RecomputeRadialReturnIsotropicPlasticity(const InputParameters & parameters) :
+    RecomputeRadialReturnStressIncrement(parameters),
+    _yield_stress_function(isParamValid("yield_stress_function") ? &getFunction("yield_stress_function") : NULL),
+    _yield_stress(getParam<Real>("yield_stress")),
+    _hardening_constant(getParam<Real>("hardening_constant")),
+    _hardening_function(isParamValid("hardening_function") ? dynamic_cast<PiecewiseLinear*>(&getFunction("hardening_function")) : NULL),
```

Comment:
I would drop the `_hardening_constant` parameter entirely and use the MOOSE capability that allows users to pass in numbers where functions are requested. This will automatically build a constant function that evaluates to teh number everywhere. It will save you all the either/or logic below.